src/scene/shader-lib/program-library.js

```diff
@@ -75,9 +75,9 @@ class ProgramLibrary {
         let def = this.definitionsCache.get(key);
         if (!def) {
             let lights;
-            if (options.lights) {
-                lights = options.lights;
-                options.lights = lights.map(function (l) {
+            if (options.litOptions?.lights) {
```

Comment:
In majority of places you use `options.litOptions.property` but here you have `options.litOptions?.property` - are you expecting litOptions to be null/undefined here but not elsewhere? and One more time few lines below.

Reply:
Yeah, because program-library isn't just called by the standard material, so the lit options member is optional.